packages/next/client/image.tsx

```diff
@@ -111,14 +112,25 @@ export type ImageProps = Omit<
   onLoadingComplete?: OnLoadingComplete
 }
 
-const {
+let {
   deviceSizes: configDeviceSizes,
   imageSizes: configImageSizes,
   loader: configLoader,
   path: configPath,
   domains: configDomains,
 } = (process.env.__NEXT_IMAGE_OPTS as any as ImageConfigComplete) ||
 imageConfigDefault
+
+function setImageConfig(imagesConfig: ImageConfigComplete) {
+  if (!imagesConfig) return
+
+  configDeviceSizes = imagesConfig.deviceSizes
+  configImageSizes = imagesConfig.imageSizes
+  configLoader = imagesConfig.loader
+  configPath = imagesConfig.path
+  configDomains = imagesConfig.domains
+}
+
 // sort smallest to largest
 const allSizes = [...configDeviceSizes, ...configImageSizes]
```

Comment:
I think you need to also assign `allSizes` again when any of the config changes.

We should probably not use globals anymore for these and instead pass the config to each helper function. We could memoize to avoid computing multiple times.

Reply:
I fixed this in b8b358f